### DISCUSSION TITLE
Controller does not visually update when using setValue

### DISCUSSION THREAD

**Discussion by AkselAllas** [react-hook-form/react-hook-form#3492] [Q&A] [answered]
https://codesandbox.io/s/agitated-surf-z2reo?file=/src/App.tsx

In the CSB, I have 2 inputs. 1st is autocomplete. 2nd is "buttonSelect", which consists of three buttons, which let you choose one value.

What I'm trying to achieve is that I can:
1) Click on autocomplete component, choose "b", and the buttonSelect component gets the button "1" selected. Then on submit I should get {"autocomplete":"b","buttonSelect":{"value":1}}
2) I can click autocomplete, choose "b", then buttonSelect "1" gets selected, but I can override the selection, and click the button "0". Then on submit I should get {"autocomplete":"b","buttonSelect":{"value":0}}

The current implementation I have in the CSB is that on line 40 I use setValue to set "buttonSelect", but the problem is that the <Controller> doesn't get rerendered and the buttonSelect doesn't get visually selected.
On submit, the form submits the correct value, but the component doesn't rerender

**Comment by AkselAllas** [comment]
My approach (not in CSB) currently uses useWatch, useEffect, setState and setting "buttonSelect" key with an uuid every click of the buttonSelect.
Very ugly, and it causes multiple warnings/errors + my second usecase doesn't work anymore, I assume because, when I set a new key for the "buttonSelect" component, the link between the rendered component and the form doesn't work anymore.

**Comment by bluebill1049** [comment, marked as answer]
I had a quick play, but the data type number and string were hard to debug and extra state.

should be able just to use setValue at `onChange`

https://codesandbox.io/s/loving-cdn-gtsd0?file=/src/App.tsx

**Comment by AkselAllas** [comment]
Thank you very much! My abstractions (which come from use with other components) made everything convoluted.

Saved my day! You're the best!

**Reply by bluebill1049** [reply]
np :) keep it simple 👍

**Reply by wowczarczyk** [reply]
Hey Bill, sorry for hijacking the thread, but my case is similar, but instead of changing the value of the buttonSelect I would like to control its `disabled` property. In other words, how to force re-render with setValue? I would like to disable one field when value of other field is changed. (Both fields are Controlled Material-UI Switches)

**Reply by wowczarczyk** [reply]
OK, I figured it out with `useWatch`. BTW,  when it comes to TS types, shouldn't the `control` in useWatch receive the `TwachFieldValues` generic type from `useWatch` like so:

```
export declare function useWatch<TWatchFieldValues extends FieldValues>(props: {
    name: string[];
    defaultValue?: UnpackNestedValue<DeepPartial<TWatchFieldValues>>;
    control?: Control<TWatchFieldValues>;
}): UnpackNestedValue<DeepPartial<TWatchFieldValues>>;
```

**Reply by bluebill1049** [reply]
hey @wowczarczyk you are right, I think the confusing right now `Controler<TFieldValues>`, it should be TWatchFieldsValue. I hope use TS 4.1, we can infer that value correctly without need `TWatchFieldsValue`
